wallet/storage/src/internal/mod.rs

```diff
@@ -31,13 +37,58 @@ use wallet_types::{
     AccountTxId, KeychainUsageState, RootKeyContent, RootKeyId, WalletTx,
 };
 
+fn password_to_sym_key(password: String, salt: &[u8; 32]) -> SymmetricKey {
```

Comment:
salt shouldn't be a parameter. It should be generated on the fly when it's needed to be used.

Reply:
it is generated on the first try, and on change password in the future. 

But for unlocking you must reuse the existing one otherwise the same password with different salt will give you a different key and we will be unable to unlock the wallet.

Reply:
In the database you should store `KdfChallenge`, which contains the salt and the other parameters you need to reconstruct the hash. It also contains the facility to compare against a password using `hash_from_challenge()`.

Reply:
fixed, changed to storing the KdfChallenge instead of just the salt.